rskj-core/src/main/java/co/rsk/peg/federation/FederationStorageProviderImpl.java

```diff
@@ -291,58 +311,67 @@ private void saveNewFederation(ActivationConfig.ForBlock activations) {
-
-        if (activations.isActive(RSKIP123)) {
-            if (oldFederation != null) {
-                saveStorageVersion(
-                    OLD_FEDERATION_FORMAT_VERSION.getKey(),
-                    oldFederation.getFormatVersion()
-                );
-            } else {
-                // assume it is a standard federation to keep backwards compatibility
-                saveStorageVersion(
-                    OLD_FEDERATION_FORMAT_VERSION.getKey(),
-                    STANDARD_MULTISIG_FEDERATION.getFormatVersion()
-                );
-            }
-            serializer = BridgeSerializationUtils::serializeFederation;
+        if (!activations.isActive(RSKIP123)) {
+            bridgeStorageAccessor.safeSaveToRepository(OLD_FEDERATION_KEY.getKey(), oldFederation, BridgeSerializationUtils::serializeFederationOnlyBtcKeys);
+            return;
         }
 
-        bridgeStorageAccessor.safeSaveToRepository(OLD_FEDERATION_KEY.getKey(), oldFederation, serializer);
+        int oldFederationFormatVersion = getOldFederationFormatVersion();
+        bridgeStorageAccessor.safeSaveToRepository(OLD_FEDERATION_KEY.getKey(), oldFederation, BridgeSerializationUtils::serializeFederation);
+        saveFederationFormatVersion(OLD_FEDERATION_FORMAT_VERSION.getKey(), oldFederationFormatVersion);
     }
 
-    // TODO
-/*    private void savePendingFederation() throws IOException {
-        if (shouldSavePendingFederation) {
-            if (activations.isActive(RSKIP123)) {
-                // we only need to save the standard part of the fed since the emergency part is constant
-                saveStorageVersion(
-                    PENDING_FEDERATION_FORMAT_VERSION.getKey(),
-                    STANDARD_MULTISIG_FEDERATION.getFormatVersion()
-                );
-            }
-            savePendingFederationToRepository(pendingFederation);
+    private int getOldFederationFormatVersion() {
+        if (oldFederation == null) {
+            // assume it is a standard federation to keep backwards compatibility
+            return STANDARD_MULTISIG_FEDERATION.getFormatVersion();
         }
-    }*/
+
+        return oldFederation.getFormatVersion();
+    }
+
+    private void savePendingFederation(ActivationConfig.ForBlock activations) {
+        if (!shouldSavePendingFederation) {
+            return;
+        }
+
+        byte[] serializedPendingFederation = serializePendingFederation(activations);
+        bridgeStorageAccessor.safeSaveToRepository(PENDING_FEDERATION_KEY.getKey(), serializedPendingFederation);
+        if (!activations.isActive(RSKIP123)) {
+            return;
+        }
+
+        // we only need to save the standard part of the fed since the emergency part is constant
+        saveFederationFormatVersion(PENDING_FEDERATION_FORMAT_VERSION.getKey(), STANDARD_MULTISIG_FEDERATION.getFormatVersion());
```

Comment:
```suggestion
        if (activations.isActive(RSKIP123)) {
            // we only need to save the standard part of the fed since the emergency part is constant
        saveFederationFormatVersion(PENDING_FEDERATION_FORMAT_VERSION.getKey(), STANDARD_MULTISIG_FEDERATION.getFormatVersion());
        }
```
Easier to read I think